ui/src/transaction.ts

```diff
@@ -210,11 +204,28 @@ function txData(txc: ContractTransaction, date: number = Date.now()): TxData {
   };
 }
 
+function registerTxUpdateCallback(tx: Tx): void {
+  if (tx.status === TxStatus.AwaitingInclusion) {
+    const provider = svelteStore.get(walletStore).provider;
+    provider.waitForTransaction(tx.hash).then(txReceipt => {
```

Comment:
We should also catch errors and log them with an appropriate error message.

Reply:
92945be8d8d15e8943902e3ef25c2c64de5d3601